src/contracts/core/PaymentCoordinatorStorage.sol

```diff
@@ -0,0 +1,105 @@
+    /// @notice The EIP-712 typehash for the contract's domain
+    bytes32 public constant DOMAIN_TYPEHASH =
+        keccak256("EIP712Domain(string name,uint256 chainId,address verifyingContract)");
+    uint64 public immutable MAX_PAYMENT_DURATION;
+    /// @notice max amount of time that a payment can start in the past
+    uint64 public immutable MAX_RETROACTIVE_LENGTH;
+    /// @notice max amount of time that a payment can start in the future
+    uint64 public immutable MAX_FUTURE_LENGTH;
+    /// @notice absolute min timestamp that a payment can start at
+    uint64 public immutable GENESIS_PAYMENT_TIMESTAMP;
+
+    /// @notice The elegationManager contract for EigenLayer
+    IDelegationManager public immutable delegationManager;
+
+    /// @notice The StrategyManager contract for EigenLayer
+    IStrategyManager public immutable strategyManager;
+
+    /*******************************************************************************
+                                       STORAGE 
+    *******************************************************************************/
+
+    /**
+     * @notice Original EIP-712 Domain separator for this contract.
+     * @dev The domain separator may change in the event of a fork that modifies the ChainID.
+     * Use the getter function `domainSeparator` to get the current domain separator for this contract.
+     */
+    bytes32 internal _DOMAIN_SEPARATOR;
+
+    /// @notice list of roots submitted by the paymentUpdater
+    DistributionRoot[] public distributionRoots;
+
+    /// Slot 3
+    /// @notice The address of the entity that can update the contract with new merkle roots
+    address public paymentUpdater;
+    /// @notice The interval in seconds at which the calculation for range payment distribution is done.
+    /// @dev Payment durations must be multiples of this interval.
+    uint64 public calculationIntervalSeconds;
+
+    /// Slot 4
+    /// @notice Delay in timestamp before a posted root can be claimed against
+    uint64 public activationDelay;
+    /// @notice the commission for all operators across all avss
+    uint16 public globalOperatorCommissionBips;
+    /// @notice Payments must start at earliest now except unless retroactive payments enabled
+    bool public retroactivePaymentsEnabled;
+
+    /// @notice Mapping: earner => the address of the entity to which new payments are directed on behalf of the earner
+    mapping(address => address) public claimerFor;
+
+    /// @notice Mapping: claimer => token => total amount claimed
+    mapping(address => mapping(IERC20 => uint256)) public cumulativeClaimed;
+
+    /// @notice Timestamp for latest submitted payment range
+    uint64 public latestPaymentRangeTimestamp;
+    /// @notice Used for unique rangePaymentHashes
+    uint256 public paymentNonce;
+    /// @notice Mapping: avs => rangePaymentHash => bool to check if range payment hash has been submitted
+    mapping(address => mapping(bytes32 => bool)) public isRangePaymentHash;
```

Comment:
we should be able to distinguish payForAll from pay in these hashes...

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/488/commits/a27911bceb4c60db7ad1a53551301dfb4615d020
Added `isRangePaymentForAllHash`